abi.html

```diff
@@ -1174,16 +1174,26 @@ <h4><a href="#vtable-components"> 2.5.2 Virtual Table Components and Order </a><
 <p>
 The order of the virtual function pointers in a virtual table is the order
 of declaration of the corresponding member functions in the class.
-There is an entry for any virtual function declared in a class,
+There is an entry for every non-<tt>consteval</tt> virtual function
+declared in a class (including deleted functions),
 whether it is a new function or overrides a base class function,
-unless it overrides a function from the primary base,
+unless it overrides a function from the primary base
 and conversion between their return types does not require an adjustment.
 (In the case of this exception,
 the primary base and the derived class share the virtual table,
 and can share the virtual function entry because their 'this' and
 result type adjustments are the same.)
-If a class has an implicitly-defined virtual destructor,
-its entries come after the declared virtual function pointers.
+If the return types are different, they are
+both pointer-to-class types, or both reference-to-class types.
+Let B and D denote the classes, where D is derived from B.  The
+conversion between the return types requires an adjustment
+if B is a morally virtual base of D 
+or if B is not located at offset zero in D.
+If a class has an implicitly-declared virtual
+copy assignment operator,
+move assignment operator, or destructor,
+those entries come after the declared virtual function pointers,
+in that order.
```

Comment:
How about:

> The order of the virtual function pointers in a virtual table is the order of declaration of the corresponding member functions in the class. If an implicitly-declared copy assignment operator, move assignment operator, or destructor is virtual, it is treated as if it were declared at the end of the class, in that order.   (Implicitly-declared assignment operators may be virtual if a base class declares a virtual assignment operator taking a reference to a derived class type.)
>
> An entry is added for every virtual function in a class, including deleted functions, unless:
> - the function is `consteval` or
> - the function overrides a function from the primary base and that override does not require a return-type adjustment.
>
> An override requires a return-type adjustment if the return types are different and have potentially incompatible representations.  C++ permits an override to differ in return type from the overridden function only if both types  are pointer-to-class or reference-to-class types and the class type `B` in the overridden function is an unambiguous base class of the class type `D` in the override.  For the purposes of v-table layout, these types are considered to have potentially-incompatible representations if:
> - `B` is a morally virtual base of `D` (even if `D` is `final` and the offset of `B` within it is known to be zero) or
> - the static offset of `B` within `D` is non-zero.

Note that one of the extensions we're considering for pointer authentication will require adding another adjustment case.

Reply:
Is there a case where an assignment operator can be simultaneously implicit, virtual, and defined as deleted, and we'd still have to add an entry for it?  That seems unfortunate, but probably not worth trying to eliminate.